### PR TITLE
Add response class for handle response objects

### PR DESCRIPTION
**Pull request by nunorafaelrocha**
* Add a message response class to handle the response objects from the http requests.
* `BitreserveClient` http methods now returns full response.
* Models now handle directly the response objects.
* Update all tests.

### CODE REVIEW

File: lib/Bitreserve/HttpClient/Message/MessageFactory.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /**
+     * Create new response.
+     *
+     * @param int $statusCode Response status code.
+     * @param array $headers Response headers.
+     * @param mixed $body Response body.
+     * @param array $options Options.
+     *
+     * @return [type] [description]
```
Comment (fixe) [inline lib/Bitreserve/HttpClient/Message/MessageFactory.php]: Lorem ipsum.

File: lib/Bitreserve/HttpClient/Message/Response.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Bitreserve\HttpClient\Message;
+
+use GuzzleHttp\Message\Response as BaseResponse;
+
+class Response extends BaseResponse
```
Comment (fixe) [inline lib/Bitreserve/HttpClient/Message/Response.php]: Missing docblock.

File: test/Bitreserve/Tests/BitreserveClientTest.php [hunk 3]
```diff
@@ -113,6 +116,8 @@ public function shouldReturnTickersByCurrency()
             'pair' => 'BTCUSD',
         ));
 
+        $stream = Stream::factory(json_encode($data));
```
Comment (fixe) [inline test/Bitreserve/Tests/BitreserveClientTest.php]: What about mocking `getContent` instead?

File: test/Bitreserve/Tests/Model/UserTest.php [hunk 4]
```diff
@@ -28,11 +30,13 @@ public function shouldPerformAUserUpdate()
     {
         $data = array('firstName' => 'foo', 'lastName' => 'bar');
 
+        $stream = Stream::factory(json_encode($data));
```
Comment (fixe) [inline test/Bitreserve/Tests/Model/UserTest.php]: Same as above.

File: test/Bitreserve/Tests/Model/TokenTest.php [hunk 5]
```diff
@@ -32,6 +34,8 @@ public function shouldReturnUser()
     {
         $data = array('username' => 'foobar');
 
+        $stream = Stream::factory(json_encode($data));
```
Comment (fixe) [inline test/Bitreserve/Tests/Model/TokenTest.php]: Same as above.

**Comment by nunorafaelrocha** [comment]
@fixe Done.

**Pull request merged**